core/src/main/java/io/grpc/DecompressorRegistry.java

```diff
@@ -0,0 +1,117 @@
+package io.grpc;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import javax.annotation.Nullable;
+
+/**
+ * Encloses classes related to the compression and decompression of messages.
+ */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/492")
+public final class DecompressorRegistry {
+
+  private static final ConcurrentMap<String, DecompressorInfo> decompressors =
```

Comment:
Maybe make all methods non-static and make a `DecompressorRegistry.getInstance()`?